contracts/RockPaperScissors.sol

```diff
@@ -0,0 +1,268 @@
+
+        game.creator = msg.sender;
+        game.stake = stake;
+        game.moveHash = playerMoveHash;
+        fundMove(stake);
+        emit GameCreated(msg.sender, gameToken, stake);
+
+        return gameToken;
+    }
+
+    /*
+    * The opponent makes their move and waits for the creator to resolve the game by revealing their move
+    */
+    function opponentPlay(bytes32 gameToken, uint8 playerMove) external payable whenNotPaused {
+
+        Game storage game = games[gameToken];
+        uint256 stake = game.stake;
+        require(stake != 0, GAME_NOT_FOUND_MSG);
+        require(game.opponent == NULL_ADDRESS, "Opponent has already moved");
+        require(playerMove < 3, INVALID_MOVE_MSG);
+
+        game.opponentMove = playerMove;
+        game.expiryDate =  block.timestamp + FORFEIT_WINDOW;
+        fundMove(stake);
+        emit OpponentPlays(msg.sender, gameToken, playerMove);
+    }
+
+
+    function fundMove(uint256 stake) internal {
+        uint256 senderBalance = balances[msg.sender];
+
+        if (msg.value < stake) {
+            require(balances[msg.sender] >= stake - msg.value, "Insufficient balance");
+            balances[msg.sender] = SafeMath.sub(senderBalance, stake - msg.value);
+            return;
+        }
+
+        balances[msg.sender] = senderBalance.add(msg.value - stake);
+        return;
+    }
+
+    /**
+    * Once the opponent has made their move, the creator of the game must resolve to determine the winner
+    * and reward the winner.
+    * If the creator does not resolve the game then the opponent can collect their winnings
+     * after the forfeit window has expired.
+    */
+    function resolveGame(bytes32 gameToken, bytes32 secret, uint8 creatorMove) external whenNotPaused {
+
+        Game storage game = games[gameToken];
+
+        address creator = game.creator;
+        require(msg.sender == creator, ONLY_CREATOR_CALL_MSG);
+        require(creatorMove < 3, INVALID_MOVE_MSG);
+
+        address opponent = game.opponent;
+        require(opponent != NULL_ADDRESS , "Still awaiting opponent");
+
+        //Validate move
+        bytes32 expectedMoveHash = hashPlayerMove(gameToken, secret, creatorMove);
+        require(game.moveHash == expectedMoveHash, "Move and secret do not match");
+
+        uint8 opponentMove = game.opponentMove;
+        uint256 stake = game.stake;
+
+        //Draw
+        if(opponentMove == creatorMove){
+            resetGame(game);
+            emit Draw(gameToken);
+            balances[creator] = balances[creator].add(stake);
+            balances[opponent] = balances[opponent].add(stake);
+            return;
+        }
+
+        //Determine winner
+        address winner;
+        address loser;
+        uint256 winnings = stake * 2;
+        uint256 fee = stake * FEE_PERCENTAGE/100;
+
+        // If the creator wins then is rewarded by not paying a fee on the winnings
+        if ((opponentMove + 1) % 3 == creatorMove) {
```

Comment:
How are you going to unit test this arithmetic resolver?

Reply:
Now you mention I think best to extract to another pure function. Too much logic in this function.